deploy/vagrant/bin/spot_request.py

```diff
@@ -203,12 +203,16 @@ def main(args):
         info('getting instance IPs...')
         for r in requests:
             instance_id = r.instance_id
+            info('waiting for ip to be allocated to the machine')
             ip = conn.get_only_instances([instance_id])[0].ip_address
```

Comment:
It should always return an array with 1 element. the instance_id is unique, and was just returned from aws, so I think it is safe to assume the machine is still up. 

Under normal circumstances, ip_address will become not None at some point. Exceptions can occur.. but I think in that case, the user can ctrl +c to stop it.